### PR TITLE
Expiration parameter for AddProvider struct (#337)

### LINKED ISSUE #337: Update `add_provider_to_msa` `AddProvider` To Have a Expiration Block
As a user, I want my `add_provider_to_msa` proof to expire in the future.

*Note:* Should follow the pattern from #336 

## Acceptance Criteria
- [ ] If the block is too FAR into the future it should be rejected
- [ ] If the block is in the past, it should be rejected

### PR DESCRIPTION
**Pull request by dopatraman**
# Goal
The goal of this PR is to check a block's expiration before adding a new provider.

with @johnmela 

Closes #337 

# Discussion
The logic for checking a block's validity is currently local to the MSA pallet. As discussed, we will table refactoring it to a common module until it is used outside the MSA pallet.

This PR also includes a script for automating JS definition generation. You can invoke it by calling

```
make js
```

# Checklist
- [x] Tests added
- [x] Weights updated

### CODE REVIEW

**Review by johnmela — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -98,6 +98,10 @@ check-rococo:
 check-mainnet:
 	SKIP_WASM_BUILD= cargo check --features  frequency
 
+.PHONY: js
+js:
+	./scripts/generate_js_definitions.sh
+
```
Comment (saraswatpuneet) [inline Makefile:104]: You should mention this in readme in js/api-augment

File: js/api-augment/README.md [hunk 2]
```diff
@@ -49,12 +49,12 @@ See [CONTRIBUTING.md](CONTRIBUTING.md) for more information.
 
 ### Updating Definitions
 
-1. Update `./definitions`
```
Comment (saraswatpuneet) [inline js/api-augment/README.md:52]: @dopatraman  I was about review 

But you should not delete point 1 , it is important as it is a manual step 

Please fix this in some other PR

**Review by saraswatpuneet — COMMENTED** [review]
@dopatraman  I will fix that in my PR

**Comment by dopatraman** [comment]
The `make` should do this automatically. People should not have to manually `cd` into a directory to generate these files, right?

**Comment by saraswatpuneet** [comment]
@dopatraman RPC methods are added to files in definition and new files are added in that which is why step 1 was added

Unless the script is somehow discovering definitions and adding new RPC or update RPC definitions then ok

**Comment by saraswatpuneet** [comment]
Step one mentions about updating code in this folder with new/update RPC calls

https://github.com/LibertyDSNP/frequency/tree/main/js/api-augment/definitions

**Comment by dopatraman** [comment]
Ok, i will add it here: https://github.com/LibertyDSNP/frequency/pull/469

**Comment by saraswatpuneet** [comment]
Thank you 💯

**Pull request merged**